business/checkers/k8sgateways/status_checker.go

```diff
@@ -0,0 +1,66 @@
+package k8sgateways
+
+import (
+	k8s_networking_v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/kiali/kiali/models"
+)
+
+type StatusChecker struct {
+	K8sGateway *k8s_networking_v1alpha2.Gateway
+}
+
+type K8sGatewayStatus struct {
+	ObjectField string
+	Status      string
+}
+
+const GwAPICheckerCode string = "GWAPI"
+
+// K8sGatewayConditionStatus represents the status failures for a Condition in a K8sGateway
+var K8sGatewayConditionStatus = map[string]string{
+	"Scheduled": "False",
+	"Ready":     "False",
+}
+
+// K8sGatewayConditionStatus represents the status failures for a Condition in a K8sGateway
+var K8sGatewayListenersStatus = map[string]string{
+	"Conflicted":   "True",
+	"Detached":     "True",
+	"Ready":        "False",
+	"ResolvedRefs": "False",
+}
+
+// Check validates that no two gateways share the same host+port combination
+func (m StatusChecker) Check() ([]*models.IstioCheck, bool) {
+	validations := make([]*models.IstioCheck, 0)
+
+	for _, c := range m.K8sGateway.Status.Conditions {
```

Comment:
It seems the 'status' field is auto-generated by Gateway API.

Reply:
Yes, the idea would be to highlight these values in the editor - like the Istio ones. Maybe there is other way to do it?